File: solutions/cpp/sum-of-multiples/1/sum_of_multiples.cpp
#include "sum_of_multiples.h"

#include <set>

using namespace std;

namespace sum_of_multiples {

    int to(set<int> factors, int limit){
        set<int> s{0};
        for(int f: factors){
            int m = f;
            while(m<limit){
                if(m!=0) 
                    s.insert(m);
                m += f;
            }
        }
        int sum = 0;
        for(int n : s){
            sum += n;
        }
        return sum;
    }
    

}  // namespace sum_of_multiples

File: solutions/julia/sum-of-multiples/1/sum-of-multiples.jl
function get_one_set(limit, factor)
    factor == 0 ? Set() : 
     Set(collect(factor:factor:limit-1))
end

function sum_of_multiples(limit, factors)
    s = Set()
    for i in 1:length(factors)
                 s = union(s, get_one_set(limit,factors[i]))
    end
    isempty(s) ? 0 : sum(s)
end
